### PR TITLE
fix: should not throw in React 19

### PR DESCRIPTION
**Pull request by ling1726**
React internals have been moved around in v19. This PR adds the React 19 internals path as a fallback.

Also updates CI to test on React 19 and adds utility to exclude tests that still fail in 19

Fixes partially #31

### CODE REVIEW

File: src/useIsStrictMode.ts [hunk 1]
```diff
@@ -3,10 +3,24 @@ import * as React from "react";
...
+    return React.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE.A.getOwner();
+  } catch {}
+
+  try {
+    // React <18
+    // @ts-ignore - using react internals
+    return React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED
+      .ReactCurrentOwner.current;
+  } catch {
+    console.error(
```
Comment (layershifter) [inline src/useIsStrictMode.ts]: Let's wrap with `process.env` check

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -26,3 +26,25 @@ jobs:
...
+        name: Checkout [main]
+        with:
+          fetch-depth: 0
+      - name: Setup node
+        uses: actions/setup-node@v3.5.1
+        with:
+          node-version: 16.x
+      - run: npm install
+      - run: npm install react@${{matrix.react}} react-dom@${{matrix.react}} --force
+      - run: npm run test:ci
```
Comment (layershifter) [inline .github/workflows/main.yml:46]: We can remove `npm run test:ci` from the main workflow now

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -26,3 +26,25 @@ jobs:
         run: |
           git status --porcelain
           git diff-index --quiet HEAD -- || exit 1
+
+  compat:
+    strategy:
+      matrix:
+        # rc tag should be react 19 release candidate
+        react: [18, rc]
```
Comment (layershifter) [inline .github/workflows/main.yml:33]: Are tests passing on 16.13 & 17? If so, let's add them

**Review by layershifter — APPROVED** [review]

File: src/useDisposable.16.17.test.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { useStrictEffect } from "./useStrictEffect";
+import { useStrictMemo } from "./useStrictMemo";
+
+vi.mock("./useStrictMemo.ts");
+vi.mock("./useStrictEffect.ts");
+
+describe("useDisposable", () => {
+  describe("in strict mode", () => {
+    it("should not call strict effect or memo", () => {
+      const factory = vi.fn().mockReturnValue(["foo", vi.fn()]);
```
Comment (ling1726) [inline src/useDisposable.16.17.test.ts:14]: in React 16 and 17 it's actually Ok for useMemo/useState to run twice. In these cases it's more worthwhile to test that `use-disposable` does not detect StrictMode at all

**Pull request merged**